### PR TITLE
make game agnostic, add luacheck, add github actions

### PR DESCRIPTION
**Pull request by wsor4035**
see title

### CODE REVIEW

**Comment by BuckarooBanzay** [comment]
didn't test but looks good to me :+1: 

also:

> local newcolor = "razzberry" -- intentionally misspelled ;-)

why was that there anyway? :thinking:

**Review by BuckarooBanzay — APPROVED** [review]

File: airbrush.lua [hunk 1]
```diff
@@ -71,12 +71,15 @@ function unifieddyes.on_airbrush(itemstack, player, pointed_thing)
 
 	local idx, hue = unifieddyes.getpaletteidx(painting_with, palette)
 	local inv = player:get_inventory()
-	if (not creative or not creative.is_enabled_for(player_name)) and not inv:contains_item("main", painting_with) then
+	if (not minetest.is_creative_enabled(player_name)) and not inv:contains_item("main", painting_with) then
```
Comment (OgelGames) [inline airbrush.lua:74]: Need to bump `min_minetest_version` to 5.3.0 for this.

File: .luacheckrc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    --luac
+    "math", "table",
+
+    -- Builtin
+    "vector", "ItemStack", "dump", "DIR_DELIM", "VoxelArea", "Settings", "PcgRandom", "VoxelManip", "PseudoRandom",
+
+    --mod produced
+
+}
```
Comment (OgelGames) [inline .luacheckrc:25]: 😉 
```suggestion
unused_args = false

globals = {
	"unifieddyes",
}

read_globals = {
	table = {fields = {"copy"}},

	"minetest", "ItemStack",
}

```

**Comment by OgelGames** [comment]
Tested this with using devtest (no dye mod), and some of the colors cannot be selected when using the airbrush, namely white, grey and black.

I think there needs to be code added to register the dyes from the `dye` mod if it's not installed.

File: .luacheckrc (hunk 2 above)
Comment (wsor4035) [reply .luacheckrc:25]: https://rubenwardy.com/minetest_modding_book/en/quality/luacheck.html#configuring-luacheck

**Comment by wsor4035** [comment]
> Tested this using devtest (no `dye` mod), and some of the colors cannot be selected when using the airbrush, namely white, grey and black.

might be related to https://github.com/mt-mods/unifieddyes/pull/5/files#diff-51f25c75e72ab95273185b8033d3f18c8d35ac5258cada768ffca5757d4c5429L90-R104 ill have to investigate further. what node did you test dying?

> I think there needs to be code added to register the dyes from the `dye` mod if it's not installed.

only whats needed should be registered, as different games may/do have there own dyes

**Comment by OgelGames** [comment]
> what node did you test dying?

Plastic (`scifi_nodes:white2`), which supports all colors.

BTW, I should mention it's not *just* white, grey and black, it's other colors too, I just didn't bother checking all 256 colors ;)

File: airbrush.lua (hunk 1 above)
Comment (wsor4035) [reply airbrush.lua:74]: https://github.com/mt-mods/unifieddyes/pull/5/commits/608c80662a006927878fce4249ff3c71157347b9

**Comment by wsor4035** [comment]
> Tested this using devtest (no `dye` mod), and some of the colors cannot be selected when using the airbrush, e.g. white, grey and black.

https://github.com/mt-mods/unifieddyes/pull/5/commits/608c80662a006927878fce4249ff3c71157347b9 resolved

**Comment by OgelGames** [comment]
That fixes it for creative, but not for survival, because you need the dye in your inventory.

IMO there are two ways to fix this: make `dye` a dependency, or register the other dyes, preferably the latter.

**Comment by wsor4035** [comment]
> IMO there are two ways to fix this: make `dye` a dependency, or register the other dyes, preferably the latter.

the point is to make it game agnostic, dye is mtg. this mod still works as intended in minetest game, and adds support for other games in creative. A additional PR can add support for survival later.

> > only whats needed should be registered, as different games may/do have there own dyes
> 
> They are needed, for crafting recipes and for using the airbrush in survival.

no, not really

**Comment by OgelGames** [comment]
> A additional PR can add support for survival later.

No, it should be added here.

**Comment by wsor4035** [comment]
> > A additional PR can add support for survival later.
> 
> No, it should be added here.

It really doesnt, as it purely doesnt allow you to use it in survival in other games. not broken/crashes/etc. so not essential

**Comment by wsor4035** [comment]
to clarify, as with any game agnostic mod, crafting of the items within it can only be done defined crafts for certain games.

**Comment by BuckarooBanzay** [comment]
Can we, if possible, continue the discussion about the overall functionality in #8 
This can be merged IMO as it just decouples the `default` dep and does add or remove functionality.
@OgelGames can you live with that?

**Comment by OgelGames** [comment]
It should be okay to leave it slightly broken for a bit, as it doesn't break anything for current users 👍

**Pull request merged**